django/middleware/csrf.py

```diff
@@ -196,6 +226,24 @@ def _set_token(self, request, response):
             # Set the Vary header since content varies with the CSRF cookie.
             patch_vary_headers(response, ('Cookie',))
 
+    def _origin_verified(self, request):
+        request_origin = request.META['HTTP_ORIGIN']
+        good_origin = '%s://%s' % (
+            'https' if request.is_secure() else 'http',
+            request.get_host(),
+        )
+        if request_origin == good_origin:
+            return True
+        if request_origin in self.allowed_origins_exact:
+            return True
+        parsed_origin = urlparse(request_origin)
```

Comment:
What should happen if this line raises an exception?

Reply:
I guess it would be better to reject the request if there's some malicious origin header that can't be parsed.

Reply:
👍  Here's a random example if you needed one:

```
>>> urlparse('http://[')
ValueError: Invalid IPv6 URL
```